### PR TITLE
Add custom tooltip for alternate chart types

### PR DESCRIPTION
**Pull request by djbarnwal**
Adds custom tooltip for all alternate chart types based on Figma mocks and discussions here - https://www.notion.so/rilldata/UXQA-Alt-Viz-fb97d9eb1c5c4fb1bfb3328e83008172?p=02801f8bb2dd4ab7aaa452dadbfd231d&pm=s

### CODE REVIEW

File: web-common/src/features/dashboards/time-dimension-details/charts/tdd-tooltip-formatter.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          <td class="color">
+            <svg width="16" height="16">
+              <circle cx="8" cy="8" r="6" style="fill:${colorMap[String(key)] || "#000"};">
+              </reccit>
+            </svg>
+          </td>
+            <td class="key">${escapeHTML(key)}</td>
+            <td class="value">${escapeHTML(String(val))}</td>
+          </tr>`;
+      }
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/time-dimension-details/charts/tdd-tooltip-formatter.ts:80]: I can probably use `new SvelteComponent` here but wondering if the abstraction justifies the performance implications.

File: web-common/src/features/charts/render/vega-tooltip.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+export class VegaLiteTooltipHandler {
+  location = "left";
+  alignment = "middle";
+  distance = 0;
+  pad = 8;
+  public valueFormatter: VLTooltipFormatter;
+
+  constructor(valueFormatter) {
+    this.valueFormatter = valueFormatter;
+    this.handleTooltip = this.handleTooltip.bind(this);
```
Comment (briangregoryholmes) [inline web-common/src/features/charts/render/vega-tooltip.ts:18]: You can skip this and just make handleTooltip an arrow function, which will also fix the linting errors when this method is used.

File: web-common/src/features/dashboards/time-dimension-details/charts/tdd-tooltip-formatter.ts (hunk 1 above)
Comment (briangregoryholmes) [reply web-common/src/features/dashboards/time-dimension-details/charts/tdd-tooltip-formatter.ts:80]: I would prefer that, but this is fine for now.

**Review by briangregoryholmes — CHANGES_REQUESTED** [review]

File: web-common/src/features/charts/render/vega-tooltip.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    const [leftPos, topPos] = placeElement({
+      location: this.location,
+      alignment: this.alignment,
+      distance: this.distance,
+      pad: this.pad,
+      parentPosition: parentBoundingClientRect,
+      elementPosition: elementBoundingClientRect,
+    });
+
+    el.setAttribute("style", `top: ${topPos}px; left: ${leftPos}px`);
```
[the remark this answers is not in the document]
Comment (djbarnwal) [reply web-common/src/features/charts/render/vega-tooltip.ts:57]: Added it to the tooltip class. That fixed the flashing issue.

**Review by briangregoryholmes — APPROVED** [review]

**Pull request merged**